docs/ref/contrib/postgres/fields.txt

```diff
@@ -458,17 +458,28 @@ using in conjunction with lookups on
 ``JSONField``
 =============
 
-.. class:: JSONField(**options)
+.. class:: JSONField(encoder=None, **options)
 
     A field for storing JSON encoded data. In Python the data is represented in
     its Python native format: dictionaries, lists, strings, numbers, booleans
     and ``None``.
-
-    If you want to store other data types, you'll need to serialize them first.
-    For example, you might cast a ``datetime`` to a string. You might also want
-    to convert the string back to a ``datetime`` when you retrieve the data
-    from the database. There are some third-party ``JSONField`` implementations
-    which do this sort of thing automatically.
+    
+    .. attribute:: encoder
+    
+        .. versionadded:: 1.11
+    
+        An optional JSON-encoding class to serialize data types not supported
+        by the standard JSON serializer (``datetime``, ``uuid``, etc.). You can
+        typically use the
+        :class:`~django.core.serializers.json.DjangoJSONEncoder` class or any
+        other :py:class:`json.JSONEncoder` subclass.
+
+        When the value is going back from the database, it will be in the format
+        chosen by the custom encoder (most often a string), so you'll need to
+        take extra steps to convert the value back to the initial data type
+        (:meth:`django.db.models.Model.from_db()` and
+        :meth:`django.db.models.Field.from_db_value` are two obvious hooks for
```

Comment:
Add some warning similar to what's at https://docs.djangoproject.com/en/1.10/topics/http/sessions/#write-your-own-serializer? (For example, you run the risk of returning a datetime that was actually a string that just happened to be in the same format chosen for datetimes).)